Python/ibmcloudsql/cos.py

```diff
@@ -0,0 +1,650 @@
+                columns={"Key": "Object"})
+        else:
+            result = pd.DataFrame()
+        return result
+
+    def get_bucket_info(self, cos_url):
+        """
+        .. code-block: console
+
+            curl https://config.cloud-object-storage.cloud.ibm.com/v1/b/my-bucket \
+            -H 'authorization: bearer <IAM_token>'
+
+        200 status code:
+
+        .. code-block:: python
+
+            {
+            "name": "my-new-bucket",
+            "crn": "crn:v1:bluemix:public:cloud-object-storage:global:a/ 3bf0d9003abfb5d29761c3e97696b71c:xxxxxxx-6c4f-4a62-a165-696756d63903:bucket:my-new-bucket",
+            "service_instance_id": "xxxxxxx-6c4f-4a62-a165-696756d63903",
+            "service_instance_crn": "crn:v1:bluemix:public:cloud-object-storage:global:a/3bf0d9003abfb5d29761c3e97696b71c:xxxxxxx-6c4f-4a62-a165-696756d63903::",
+            "time_created": "2018-03-26T16:23:36.980Z",
+            "time_updated": "2018-10-17T19:29:10.117Z",
+            "object_count": 764265234,
+            "bytes_used": 28198745752445144
+            }
+
+        todo
+        ----
+            https://cloud.ibm.com/apidocs/cos/cos-configuration?code=python
+        """
+        from cos_config.resource_configuration_v1 import ResourceConfigurationV1
+        bucket = self.get_bucket(cos_url)
+
+        if bucket in self.buckets_info:
+            return self.buckets_info[bucket]
+        else:
+            client = ResourceConfigurationV1(iam_apikey=self.apikey)
+            config = client.get_bucket_config(bucket)
+            self.buckets_info[bucket] = config
+            return config
+        # bucket = ParsedUrl().get_bucket(cos_url)
+        # response = requests.get(
+        #     "https://config.cloud-object-storage.cloud.ibm.com/v1/b/{bucket}".
+        #     format(bucket=bucket),
+        #     headers=self.request_headers,
+        # )
+
+        # if response.status_code == 200 or response.status_code == 201:
+        #     status_response = response.json()
+        #     jobStatus = status_response['status']
+        #     if jobStatus == 'completed':
+        #         break
+        #     if jobStatus == 'failed':
+        #         print("Job {} has failed".format(jobId))
+        #         break
+        # else:
+        #     print("Job status check failed with http code {}".format(
+        #         response.status_code))
+        #     break
+        # time.sleep(2)
+
+    def update_bucket(self, cos_url):
+        """
+        todo
+        ----
+           revise this
+        """
+        from cos_config.resource_configuration_v1 import ResourceConfigurationV1
+
+        api_key = self.apikey
+        bucket = self.get_bucket(cos_url)
+
+        client = ResourceConfigurationV1(iam_apikey=api_key)
+
+        client.update_bucket_config(
+            bucket,
+            firewall={"allowed_ip": ["10.142.175.0/22", "10.198.243.79"]})
+
+    def connect_project_lib(self, project, file_name=None):
```

Comment:
I assume the "project_lib" is the Watson Studio project Python client, isn't it? Since ibmcloudsql can be used in any Python runtime it is not clear to the users that this here is for Watson Studio environments specifically. So please make sure it is at least explained in the code and documentation accordingly that project_lib comes from Watson Studio. I am wondering whether we should maybe even call that out in the variable name and method signature instead of just generic "project_lib" naming.